number examples

maboa opened the pull request:
Ready for review. Thanks :)

schalkneethling commented on live-examples/js-examples/number-positive-infinity.html:
```diff
@@ -0,0 +1,15 @@
+<pre>
+<code id="static-js">function checkNumber(bigNumber) {
+	if (bigNumber === Number.POSITIVE_INFINITY) { return "Process number as Infinity" };
```
I know we have not decided on a style but, here I reckon the line length itself presents a problem and writing this as, would be preferable:

```
if (bigNumber === Number.POSITIVE_INFINITY) {
    return "Process number as Infinity"
};
```

maboa replied:
Yup.

schalkneethling commented on live-examples/js-examples/number-negative-infinity.html:
```diff
@@ -0,0 +1,15 @@
+  	return smallNumber;
+}
+
+var x = -Number.MAX_VALUE, x2 = -Number.MAX_VALUE * 2;
```
My suggestion here would be to have two explicit `var` declarations. I would say this is a style rule we can agree to but, also for consistency with other code.

```
var x = -Number.MAX_VALUE;
var x2 = -Number.MAX_VALUE * 2;
```

maboa replied:
OK - yes, just trying to space-save again. But I think we've established that consistency is more important.

schalkneethling commented on live-examples/js-examples/number-min-value.html:
```diff
@@ -0,0 +1,13 @@
+<pre>
+<code id="static-js">function multiply(x,y) {
+  if (x * y < Number.MIN_VALUE) { return ("Process as -Infinity")};
```
Any specific reason the string is wrapped in parentheses?

maboa replied:
Nope - just a mistake :) We should also maybe think about where we use double quotes as opposed to single. For example airbnb's JS code style is to have them in single quotes: https://github.com/airbnb/javascript#strings

schalkneethling replied:
Agree, I definitely vote for single quotes over double.

schalkneethling commented on live-examples/js-examples/number-max-value.html:
```diff
@@ -0,0 +1,13 @@
+<pre>
+<code id="static-js">function multiply(x,y) {
+  if (x * y > Number.MAX_VALUE) { return ("Process as Infinity")};
```
Same here re: parentheses

schalkneethling commented on live-examples/js-examples/number-issafeinteger.html:
```diff
@@ -0,0 +1,13 @@
+<pre>
+<code id="static-js">function warn(x) {
+  	if (Number.isSafeInteger(x)) return "Precision safe.";
```
Missing opening and closing curly braces.

maboa replied:
OK - yes. Let's stick to that.

schalkneethling commented on live-examples/js-examples/number-isnan.html:
```diff
@@ -0,0 +1,14 @@
+<pre>
+<code id="static-js">function typeOfNaN(x) {
+  if (Number.isNaN(x)) return "Number NaN";
+	if (isNaN(x)) return "NaN";
```
Nit: Missing curly braces and inconsistent indentation.

schalkneethling commented on live-examples/js-examples/number-isinteger.html:
```diff
@@ -0,0 +1,13 @@
+<pre>
+<code id="static-js">function fits(x, y) {
+  if (Number.isInteger(y / x)) return "Fits!";
```
Nit: Missing curly braces

maboa replied:
OK

schalkneethling commented on live-examples/js-examples/number-epsilon.html:
```diff
@@ -0,0 +1,14 @@
+<pre>
+<code id="static-js">x = 0.2;
+y = 0.3;
+z = 0.1;
```
Missing `var` for the above three lines.

maboa replied:
OK

maboa commented:
I have hopefully addressed most of the issues in the latest commit.

I wonder whether the idea of trying to create functions that do something potentially useful is a good one. What I'm trying to avoid is just having a series of `console.log`s as is often presented in page, preferring to show stuff actually being used.

For example I'd be interested if people prefer the current `Number.EPSILON` example https://github.com/mdn/interactive-examples/pull/276/files#diff-bcbec236e9a5fe8be6c3e53cd60a90c1 or perhaps an alternative more like the one below? @wbamberg @schalkneethling @Elchi3 
```
function correct(x) {
  if (x < Number.EPSILON) {
    return 0;
  }
  return x;
}

console.log(0.2 - 0.3 + 0.1);
// expected output: 2.7755575615628914e-17

console.log(correct(0.2 - 0.3 + 0.1));
// expected output: 0
```

Elchi3 commented:
> I wonder whether the idea of trying to create functions that do something potentially useful is a good one. 

I think (wrapper) functions are not necessarily helpful. I see the examples we are creating here as code snippets that you can quickly glance at and possibly copy-paste fragments of. Imo, a function adds complexity as they make you wonder what they are for, which isn't necessarily needed for just exploring. Also, when playing, you are interacting with the wrapper like `correct(foo)` and not with the API in question (`Number.EPSILON`) directly. I think there might be cases where this is fine and definitely the Number and Math functions are often appearing in utility functions that actually do something useful like a mathematical operation. On the other side, your wrapper function for `toLocaleString` here, for example, doesn't add much value for me. It really is just a wrapper.

I'm trying to deep dive, so that my feedback here becomes more clear: In the `toString` example, the `hexColor` function (we're using en-US spelling on MDN btw), is a really nice one to show the usefulness of the radix parameter. As a complete beginner, or someone who knows nothing about hex values, it might take a moment to understand how this is actually done (my thought process was like: "ok, there is a function, then there is an if-condition, ah, there is `Math.abs`, hm probably not important, ah, there is the `toString`, and oh it has a parameter, that must do the core thing here). So, now I know how to calculate hex values and that `toString` has a parameter. If I want to play with `toString()` as I have a different use case, I will have to tweak the `hexColor` function or add my own plain `toString` calls. What I want to outline here is that I think we should be careful with what kind of example-case we're using, as it can add complexity to understanding the API (the reader has to think this through, somehow like I did). I think this complexity what @wbamberg meant in https://github.com/mdn/interactive-examples/pull/271#pullrequestreview-61950345 as well. I'm not saying we shouldn't add any additional code/functions/complexity to make the examples practical and worthwhile, I just think we have to be careful with the level of complexity and keep in mind that the API might be used differently and that the wrapping isn't unnecessarily hiding the API we want to illustrate.

schalkneethling commented:
@maboa I believe you need to squash your commits and rebase against master to get rid of the conflict. Other than that, it seems all good.

RE: single vs double quotes - My vote would be for single quotes.
